src/handlers/miner_discovery_handler.erl

```diff
@@ -0,0 +1,123 @@
+-module(miner_discovery_handler).
+
+-behavior(libp2p_framed_stream).
+
+-include_lib("helium_proto/include/discovery_pb.hrl").
+
+%% ------------------------------------------------------------------
+%% API Function Exports
+%% ------------------------------------------------------------------
+
+-export([
+    server/4,
+    client/2,
+    add_stream_handler/1
+]).
+
+%% ------------------------------------------------------------------
+%% libp2p_framed_stream Function Exports
+%% ------------------------------------------------------------------
+-export([
+    init/3,
+    handle_data/3,
+    handle_info/3
+]).
+
+-define(VERSION, "discovery/1.0.0").
+
+-record(state, {}).
+
+%% ------------------------------------------------------------------
+%% API Function Definitions
+%% ------------------------------------------------------------------
+server(Connection, Path, _TID, Args) ->
+    libp2p_framed_stream:server(?MODULE, Connection, [Path | Args]).
+
+client(Connection, Args) ->
+    libp2p_framed_stream:client(?MODULE, Connection, Args).
+
+-spec add_stream_handler(pid() | ets:tab()) -> ok.
+add_stream_handler(SwarmTID) ->
+    libp2p_swarm:add_stream_handler(
+        SwarmTID,
+        ?VERSION,
+        {libp2p_framed_stream, server, [?MODULE, self(), SwarmTID]}
+    ).
+
+%% ------------------------------------------------------------------
+%% libp2p_framed_stream Function Definitions
+%% ------------------------------------------------------------------
+
+init(client, _Conn, _Args) ->
+    lager:info("client started with ~p", [_Args]),
+    {ok, #state{}};
+init(server, _Conn, _Args) ->
+    lager:info("server started with ~p", [_Args]),
+    {ok, #state{}}.
+
+handle_data(server, Data, State) ->
+    lager:info("got ~p ~p", [Data, State]),
+    #discovery_start_pb{
+        hotspot = HotspotPubkeyBin,
+        packets = Packets,
+        signature = Sig
+    } = discovery_pb:decode_msg(Data, discovery_start_pb),
+    Hotspot = erlang:list_to_binary(libp2p_crypto:bin_to_b58(HotspotPubkeyBin)),
```

Comment:
Seems a little odd that we're converting a binary to a binary again? If there's a particular reason to do so, maybe add a comment here?